charts/nr-k8s-otel-collector/templates/daemonset-configmap.yaml

```diff
@@ -0,0 +1,289 @@
+apiVersion: v1 
+kind: ConfigMap
+metadata:
+  name: {{ include "newrelic.common.naming.fullname" . }}-daemonset-config
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+data:
+  otel-collector-config.yaml: |
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+          http:
+      hostmetrics:  
+        collection_interval: 20s
+        scrapers:
+          cpu:
+            metrics:
+              system.cpu.time:
+                enabled: false
+              system.cpu.utilization:
+                enabled: true
+          load:
+          memory:
+            metrics:
+              system.memory.utilization:
+                enabled: true
+          paging:
+            metrics:
+              system.paging.utilization:
+                enabled: false
+              system.paging.faults:
+                enabled: false
+          filesystem:
+            metrics:
+              system.filesystem.utilization:
+                enabled: true
+          disk:
+            metrics:
+              system.disk.merged:
+                enabled: false
+              system.disk.pending_operations:
+                enabled: false
+              system.disk.weighted_io_time:
+                enabled: false
+          network:
+            metrics:
+              system.network.connections:
+                enabled: false
+          processes:
+          process:
+            metrics:
+              process.cpu.utilization:
+                enabled: true
+              process.cpu.time:
+                enabled: false
+      kubeletstats:
+        collection_interval: 20s
+        auth_type: "serviceAccount"
+        endpoint: "${KUBE_NODE_NAME}:10250"
```

Comment:
If you take a look at [line 94](https://github.com/newrelic/helm-charts/pull/1342/files#diff-b9dcfedad3cb705d7e2b97544a9c7802a11e8c32eb5113ed4ae65cbc7b176901R94) you'll see that the exporter goes to the kubelet using the API server as proxy in case there are firewall rules that forbid to connect directly to the node.

You should parametrize this even if the default is "`NODE_IP`".

Reply:
can you clarify what you mean by 'parametrize this even if the default is "NODE_IP"' ? 

note: we are trying to remove kubeletstats in a followup PR, but I would appreciate the clarification regardless for my understanding :)

Reply:
Take a look at the connector of `nri-kubernetes`: https://github.com/newrelic/nri-kubernetes/blob/a4bdffd9d96b446c4fa7ba8351f981656b3b091d/src/kubelet/client/connector.go#L56-L91

There are cases were security policies do not allow a pod to connect to the node directly to scrape Kubelet. So the connector fallbacks from using `$NODE_IP:10250` to `$APISERVER/api/v1/nodes/%s/proxy/`.

This is why the endpoint should be parametric somehow.

Reply:
I would leave this to a follow up PR but leave a huge comment because this could be a huge issue if this goes GA.